Increase the minimum supported Kubernetes version to v1.19

hjiajing opened the pull request:
Since Kubernetes v1.19, some APIs are deprecated. Add deprecated comment to those APIs and update documents.

Reference: https://kubernetes.io/docs/reference/using-api/deprecation-guide/

luolanzone commented on README.md:
```diff
@@ -27,7 +27,7 @@ Network Policies in a very efficient manner.
 ## Prerequisites
 
-Antrea has been tested with Kubernetes clusters running version 1.16 or later.
+Antrea has been tested with Kubernetes clusters running version 1.19 or later.
```
Hi @hjiajing you misunderstood the requirement, you need to input a field like 'deprecated: true' to our old CRDs version. Please check [here](https://github.com/kubernetes/kubernetes/blob/3ec6a387955b1240ad6d795663513f1ee12ceaec/cluster/addons/volumesnapshots/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml#L97)

hjiajing replied:
Thanks for the reminder, I will add these fields soon.

luolanzone commented on build/charts/antrea/crds/externalentity.yaml:
```diff
@@ -44,6 +44,9 @@ spec:
                 externalNode:
                   type: string
     - name: v1alpha1
+      deprecated: true
```
There are a few more CRDs with two or more served versions, I think we need to add this field for all old versions. @antoninbas what's your thoughts?

antoninbas replied:
Yes, we have more deprecated CRD versions according to https://github.com/antrea-io/antrea/blob/main/docs/api.md#crds-in-crdantreaio

Looking at this document, I have 2 questions though:
- why is `v1alpha1/ExternalEntity` not listed in the doc?
- the doc mentions that `v1alpha1/Tier` is supposed to be removed in Antrea v2.0. Are we still planning to do that?

hjiajing replied:
> There are a few more CRDs with two or more served versions, I think we need to add this field for all old versions. @antoninbas what's your thoughts?

Yes, that is what I want to ask. So, if one CRD has more than one served version, we just keep the latest one and add `deprecated: true` and deprecation warning to other versions?.

antoninbas replied:
Yes. https://github.com/antrea-io/antrea/blob/main/docs/api.md#crds-in-crdantreaio should also be up-to-date in that regard (see `Deprecated in` column).

hjiajing replied:
Updated.

antoninbas replied:
Pinging @luolanzone for the questions above: https://github.com/antrea-io/antrea/pull/6089#discussion_r1539749892

hjiajing replied:
I noticed that `v1alpha1/ExternalEntity` is not served, so can we just remove this APIVersion, cause the `deprecated` filed will not work on this CRD.

luolanzone replied:
I checked the doc history, seems we didn't add  v1alpha1/ExternalEntity at the very beginning in this doc.
As @hjiajing  mentioned, it's not served and I can only see following content in crds.yml without any schema. I believe we can simply remove it in this PR.
```
    - name: v1alpha1
      served: false
      storage: false
      schema:
        openAPIV3Schema:
          type: object
```
For `v1alpha1/Tier`, I think we can remove it. I will handle this.

luolanzone replied:
The PR is submitted to remove the v1alpha1/Tier https://github.com/antrea-io/antrea/pull/6162
@hjiajing you need to rebase after the merge. Thanks.

hjiajing replied:
Thanks for the reminder, I will rebase after that PR is merged.

hjiajing replied:
Hi @antoninbas , I created a PR to remove this CRD https://github.com/antrea-io/antrea/pull/6177

antoninbas replied:
Thanks. I assume this PR will need to be rebased after we merge #6177

luolanzone reviewed (commented):
LGTM overall, @hjiajing  could you check the comment of PR6177, so we can merge it and move forward for this PR? Thanks.

hjiajing commented:
> LGTM overall, @hjiajing could you check the comment of PR6177, so we can merge it and move forward for this PR? Thanks.

Sure, I have addressed the comment of #6177 , I will move forward after that PR is merged.

hjiajing commented:
Hi @luolanzone @antoninbas , I have rebased to the latest main branch, could you please help to check this PR, thanks.

antoninbas commented on build/charts/antrea/crds/clusternetworkpolicy.yaml:
```diff
@@ -10,6 +10,8 @@ spec:
     - name: v1alpha1
       served: true
       storage: false
+      deprecated: true
+      deprecationWarning: "crd.antrea.io/v1alpha1 ClusterNetworkPolicy is deprecated; use crd.antrea.io/v1beta1 ClusterNetworkPolicy"
```
this API (and some other ones) were supposed to be removed in #6162 
I think @luolanzone may have missed some files when updating her PR
I will open a PR to address this

antoninbas replied:
If #6238 and #6239 are merged, this is likely to be just a documentation PR, as all deprecated APIs will have been removed.
However, we will be able to use these attributes (`deprecated` / `deprecationWarning`) in the future, which is a good thing.

luolanzone commented:
@hjiajing  please update the PR message to resolve issue #5879

antoninbas commented:
/skip-all
